tests/Tests.AzureAppConfiguration/PushRefreshTests.cs

```diff
@@ -0,0 +1,302 @@
+			Assert.False(IsPushNotificationValid(pushNotification));
+			Assert.False(EventGridEventParser.TryParseJson(nullMessage, out pushNotification));
+			Assert.False(IsPushNotificationValid(pushNotification));
+
+			//These Assertions should return false since a parameter is missing from the message
+			Assert.False(EventGridEventParser.TryParseJson(noSyncTokenMsg, out pushNotification));
+			Assert.False(IsPushNotificationValid(pushNotification));
+			Assert.False(EventGridEventParser.TryParseJson(noUriMsg, out pushNotification));
+			Assert.False(IsPushNotificationValid(pushNotification));
+			Assert.False(EventGridEventParser.TryParseJson(noEventTypeMsg, out pushNotification));
+			Assert.False(IsPushNotificationValid(pushNotification));
+		}
+
+		[Fact]
+		public void ProcessPushNotificationThrowsArgumentExceptions()
+        {
+			var mockResponse = new Mock<Response>();
+			var mockClient = GetMockConfigurationClient();
+
+			IConfigurationRefresher refresher = null;
+
+			var config = new ConfigurationBuilder()
+				.AddAzureAppConfiguration(options =>
+				{
+					options.Client = mockClient.Object;
+					options.Select("*");
+					options.ConfigureRefresh(refreshOptions =>
+					{
+						refreshOptions.Register("TestKey1", "label")
+							.SetCacheExpiration(TimeSpan.FromDays(30));
+					});
+					refresher = options.GetRefresher();
+				})
+				.Build();
+
+			foreach (PushNotification invalidPushNotification in invalidPushNotificationList)
+            {
+				Action action = () => refresher.ProcessPushNotification(invalidPushNotification);
+				Assert.Throws<ArgumentException>(action); 
+			}
+
+			PushNotification nullPushNotification = null;
+
+			Action nullAction = () => refresher.ProcessPushNotification(nullPushNotification);
+			Assert.Throws<ArgumentNullException>(nullAction);
+		}
+
+		[Fact]
+		public void SyncTokenUpdatesCorrectNumberOfTimes()
+		{
+			// Arrange
+			var mockResponse = new Mock<Response>();
+			var mockClient = GetMockConfigurationClient();
+
+			IConfigurationRefresher refresher = null;
+
+			var config = new ConfigurationBuilder()
+				.AddAzureAppConfiguration(options =>
+				{
+					options.Client = mockClient.Object;
+					options.Select("*");
+					options.ConfigureRefresh(refreshOptions =>
+					{
+						refreshOptions.Register("TestKey1", "label")
+							.SetCacheExpiration(TimeSpan.FromDays(30));
+					});
+					refresher = options.GetRefresher();
+				})
+				.Build();
+
+			List<PushNotification> pushNotifications = pushNotificationList;
+
+			foreach (PushNotification pushNotification in pushNotifications)
+			{
+				refresher.ProcessPushNotification(pushNotification, TimeSpan.FromSeconds(0));
+				refresher.RefreshAsync().Wait();
+			}
+
+			mockClient.Verify(c => c.GetConfigurationSettingAsync(It.IsAny<ConfigurationSetting>(), It.IsAny<bool>(), It.IsAny<CancellationToken>()), Times.Exactly(8));
+			mockClient.Verify(c => c.UpdateSyncToken(It.IsAny<string>()), Times.Exactly(8));
+		}
+
```

Comment:
Can you a test for validating that push refresh flow actually refreshes configuration? Set a high cache expiration time, process push notification, verify that next RefreshAsync call updates config.